baybe/targets/binary.py

```diff
@@ -0,0 +1,91 @@
+"""Binary targets."""
+
+from typing import TypeAlias
+
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe.exceptions import InvalidTargetValueError
+from baybe.serialization import SerialMixin
+from baybe.targets.base import Target
+
+ChoiceValue: TypeAlias = bool | int | float | str
+"""Types of values that a :class:`BinaryTarget` can take."""
+
+_POSITIVE_VALUE_COMP = True
+"""Computational representation of the positive choice value."""
+
+_NEGATIVE_VALUE_COMP = False
+"""Computational representation of the negative choice value."""
+
+
+@define(frozen=True)
+class BinaryTarget(Target, SerialMixin):
+    """Class for binary targets."""
+
+    # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1336
+
+    positive_value: ChoiceValue = field(
+        default=1,
+        validator=instance_of(ChoiceValue),  # type: ignore[call-overload]
+        kw_only=True,
+    )
+    """Experimental representation of the positive value."""
+
+    negative_value: ChoiceValue = field(
+        default=0,
+        validator=instance_of(ChoiceValue),  # type: ignore[call-overload]
+        kw_only=True,
+    )
+    """Experimental representation of the negative value."""
+
+    @negative_value.validator
+    def _validate_values(self, _, value):
+        """Validate that the two choice values of the target are different."""
+        if value == self.positive_value:
+            raise ValueError(
+                f"The two choice values of a '{BinaryTarget.__name__}' must be "
+                f"different but the following value was provided for both choices of "
+                f"target '{self.name}': {value}"
+            )
+
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:  # noqa: D102
+        # TODO: The method (signature) needs to be refactored, potentially when
+        #   enabling multi-target settings. The current input type suggests that passing
+        #   dataframes is allowed, but the code was designed for single targets and
+        #   desirability objectives, where only one column is present.
+        assert data.shape[1] == 1
+
+        # Validate target values
+        col = data.iloc[:, [0]]
+        invalid = col[~col.isin([self.positive_value, self.negative_value]).values]
```

Comment:
No, don't think so. That would just open doors for silent errors. The binary target is – as the name suggests – a binary object. Just because we encode the two choice values with arbitrary objects, there is no closeness or whatsoever associated. And now that I see the code again, I'm even tempted to change the default to `True`/`False`? What do you think, @AVHopp @Scienfitz? Would reflect the binary nature better

Reply:
+1 for `True`/`False`

Reply:
changed